controllers/apps/transformer_component_workload.go

```diff
@@ -597,11 +606,84 @@ func (r *componentWorkloadOps) scaleOut(itsObj *workloads.InstanceSet) error {
 	}
 }
 
+func (r *componentWorkloadOps) annotateInstanceSetForMemberJoin() {
+	if r.synthesizeComp.LifecycleActions.MemberJoin == nil {
+		return
+	}
+
+	podsToMemberjoin := getPodsToMemberJoinFromAnno(r.runningITS)
+
+	for podName := range r.desiredCompPodNameSet {
+		if r.runningItsPodNameSet.Has(podName) {
```

Comment:
r.runningItsPodNameSet refers only to the members that the running instance has, rather than to all members that have joined.
If one scale-out occurs while another is still in progress, it seems to overwrite the first one, preventing it from ever being completed.